rustls/src/server/tls13.rs

```diff
@@ -158,6 +159,26 @@ mod client_hello {
                 ));
             }
 
+            if client_hello.has_certificate_compression_extension_with_duplicates() {
+                return Err(cx.common.send_fatal_alert(
+                    AlertDescription::IllegalParameter,
+                    PeerMisbehaved::OfferedDuplicateCertificateCompressions,
+                ));
+            }
+
+            let cert_compressor =
+                if let Some(offered) = client_hello.certificate_compression_extension() {
+                    // prefer server order when choosing a compression: the client's
+                    // extension here does not denote any preference.
+                    self.config
+                        .cert_compressors
+                        .iter()
+                        .find(|compressor| offered.contains(&compressor.algorithm()))
+                        .cloned()
+                } else {
+                    None
+                };
```

Comment:
I think you can drop the `else` and simplify this a bit chaining an `and_then()` off of the extension optional:

```rust
 let cert_compressor = client_hello.certificate_compression_extension().and_then(|offered|
                // prefer server order when choosing a compression: the client's
                // extension here does not denote any preference.
                self.config
                    .cert_compressors
                    .iter()
                    .find(|compressor| offered.contains(&compressor.algorithm()))
                    .cloned()
            );

```